catalyst_voices_packages/catalyst_cardano_serialization/lib/src/scripts.dart

```diff
@@ -0,0 +1,497 @@
+      _invalidCborError(value);
+    }
+  }
+
+  /// Checks if a CBOR value is a valid list-based native script.
+  static void _checkListNativeScriptValidity(CborValue value) {
+    if (value is! CborList || value.length != 2 || value[1] is! CborList) {
+      _invalidCborError(value);
+    }
+  }
+
+  /// Checks if a CBOR value is a valid bounded native script.
+  static void _checkBoundedNativeScriptValidity(CborValue value) {
+    if (value is! CborList || value.length != 2 || value[1] is! CborSmallInt) {
+      _invalidCborError(value);
+    }
+  }
+}
+
+/// Class representing a public key based native script.
+class ScriptPubkey extends NativeScript {
+  /// Public key hash.
+  final Uint8List addrKeyhash;
+
+  /// Constructor for the [ScriptPubkey] class.
+  const ScriptPubkey(this.addrKeyhash);
+
+  /// Converts the [ScriptPubkey] to its CBOR format.
+  @override
+  CborValue toCbor() => CborList(
+        [CborSmallInt(NativeScriptType.pubkey.value), CborBytes(addrKeyhash)],
+      );
+
+  /// Factory constructor to create a [ScriptPubkey] from a CBOR list.
+  factory ScriptPubkey.fromCbor(CborList value) {
+    NativeScript._checkGenericNativeScriptValidity(value);
+    return ScriptPubkey(Uint8List.fromList((value[1] as CborBytes).bytes));
+  }
+
+  /// Equatable props for value comparison.
+  @override
+  List<Object?> get props => [addrKeyhash];
+}
+
+/// Class representing an "all" native script (AND operation).
+class ScriptAll extends NativeScript {
+  /// List of the required signatures.
+  final List<NativeScript> nativeScripts;
+
+  /// Constructor for the [ScriptAll] class.
+  const ScriptAll(this.nativeScripts);
+
+  /// Converts the [ScriptAll] to its CBOR format.
+  @override
+  CborValue toCbor() => CborList([
+        CborSmallInt(NativeScriptType.all.value),
+        CborList(nativeScripts.map((s) => s.toCbor()).toList()),
+      ]);
+
+  /// Factory constructor to create a [ScriptAll] from a CBOR list.
+  factory ScriptAll.fromCbor(CborList value) {
+    NativeScript._checkListNativeScriptValidity(value);
+    final scripts = (value[1] as CborList)
+        .map((e) => NativeScript.fromCbor(e as CborList))
+        .toList();
+    return ScriptAll(scripts);
+  }
+
+  /// Equatable props for value comparison.
+  @override
+  List<Object?> get props => [nativeScripts];
+}
+
+/// Class representing an "any" native script (OR operation).
+class ScriptAny extends NativeScript {
+  /// List of the available signatures.
+  final List<NativeScript> nativeScripts;
+
+  /// Constructor for the [ScriptAny] class.
+  const ScriptAny(this.nativeScripts);
+
+  /// Converts the [ScriptAny] to its CBOR format.
+  @override
+  CborValue toCbor() => CborList([
+        CborSmallInt(NativeScriptType.any.value),
+        CborList(nativeScripts.map((s) => s.toCbor()).toList()),
+      ]);
+
+  /// Factory constructor to create a [ScriptAny] from a CBOR value.
+  factory ScriptAny.fromCbor(CborValue value) {
```

Comment:
Consider to keep `fromCbor` constructor before `toCbor`. Dart convention which we're following requires putting constructors before methods.

Please review other changes in the PR for this pattern.